src/PluginManagerAlter/Microsoft.Performance.Toolkit.PluginManager.Core.Alter.NuGet/NuGetPluginDiscovererProvider.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Performance.Toolkit.PluginManager.Core.Alter.Discovery;
+using Microsoft.Performance.Toolkit.PluginManager.Core.Alter.Credential;
+using NuGet.Configuration;
+
+namespace Microsoft.Performance.Toolkit.PluginManager.Core.Alter.NuGet
+{
+    public class NuGetPluginDiscovererProvider : IPluginDiscovererProvider
+    {
+        private Lazy<ICredentialProvider> credentialProvider =
+            new Lazy<ICredentialProvider>(() => new NuGetCredentialProvider());
+
+        public Lazy<ICredentialProvider> CredentialProvider
+        {
+            get
+            {
+                return this.credentialProvider;
+            }
+        }
+
+        public IPluginDiscoverer CreateDiscoverer(PluginSource source)
+        {
+            throw new NotImplementedException();
```

Comment:
I assume this should return a new NuGetPluginDiscoverer

Reply:
Updated.